File: airline/src/main/java/com/airline/inventory/domain/Seat.java
package com.airline.inventory.domain;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Seat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long flightId;

    private String seatNumber;

    @Enumerated(EnumType.STRING)
    private SeatStatus status;

    @Version
    private Long version;
}

File: airline_app/lib/models/seat.dart
class Seat {
  final int id;
  final int flightId;
  final String seatNumber;
  final String status;

  Seat({
    required this.id,
    required this.flightId,
    required this.seatNumber,
    required this.status,
  });

  factory Seat.fromJson(Map<String, dynamic> json) {
    return Seat(
      id: json['id'],
      flightId: json['flightId'],
      seatNumber: json['seatNumber'],
      status: json['status'],
    );
  }
}
